src/deprecations.jl

```diff
@@ -0,0 +1 @@
+Base.@deprecate copy!(dest::AWSCredentials, src::AWSCredentials) copyto!(dest, src)
```

Comment:
Should add a comment indicating when this deprecation was added so it's clear when it should be removed.